version: '3'

services:
  mbp_chatbot_db:
    image: postgres:latest
    restart: always
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    ports:
      - '5432:5432'
    volumes:
      - mbp_chatbot_db:/var/lib/postgresql/data
    networks:
      - rabbitmq_go_net # Added to the same network

  rabbitmq:
    image: rabbitmq:3-management-alpine
    container_name: 'rabbitmq'
    ports:
      - 5672:5672
      - 15672:15672
    volumes:
      - ~/.docker-conf/rabbitmq/data/:/var/lib/rabbitmq/
      - ~/.docker-conf/rabbitmq/log/:/var/log/rabbitmq
    networks:
      - rabbitmq_go_net

  bot-app:
    build: .
    ports:
      - '3001:3001'
    depends_on:
      - mbp_chatbot_db
      - rabbitmq
    environment:
      - DATABASE_URL=postgresql://postgres:postgres@mbp_chatbot_db:5432/postgres # Updated to use service name
      - RABBITMQ_URL=amqp://rabbitmq:5672
    networks:
      - rabbitmq_go_net

volumes:
  mbp_chatbot_db:
    driver: local

networks:
  rabbitmq_go_net:
    driver: bridge
